okhttp-tests/src/test/java/okhttp3/EventListenerTest.java

```diff
@@ -611,88 +613,175 @@ private void enableTlsWithTunnel(boolean tunnelProxy) {
     server.useHttps(sslClient.socketFactory, tunnelProxy);
   }
 
-  static final class DnsStart {
+  static class CallEvent {
```

Comment:
There is no loss of type information as the fields are still toplevel.

The intent here was to implement as many of the asserts of matching pairs as part of the listener impl.  I'd like connectionReleased to fail fast if a matching connectionAcquired isn't already in place.  My goal was to incrementally implement the rules of the docs to make each new (edge case) unit test trivial to read and understand while having strong asserts.